### PR TITLE
bsp: Added support for IDF v4.4 into ESP-BOX

### PR DESCRIPTION
**Pull request by espzav**
# ESP-BSP Pull Request checklist

- [x] Version of modified component bumped
- [x] CI passing

# Change description
- Added support for IDF v4.4 into ESP-BOX
- Fixed display_audio_photo example (recording)
- Updated Pull Request template for new BSPs
- Used new API for button handling
- New API for display init with configuration
- Removed minimal IDF version from examples

Closes https://github.com/espressif/esp-bsp/issues/151

### CODE REVIEW

**Comment by VojtechBartoska** [comment]
Hello, @leeebo & @espressif2022, can you please help with review? Thanks

**Comment by espzav** [comment]
I changed to draft now, because I will changing the buttons API to new one and I should change CI for test examples for IDF4.4 too.

File: examples/display_audio_photo/main/idf_component.yml
Comment (leeebo) [inline examples/display_audio_photo/main/idf_component.yml:6]: To make sure the component manager resolve a correct version:

```suggestion
    version: ">=2.5.0"
```

**Comment by leeebo** [comment]
LGTM! Just leave some comments from my point of view

**Comment by wujiangang** [comment]
@donghengqaz Please check whether this PR meets your requirements in https://github.com/espressif/esp-bsp/issues/151.

**Comment by espzav** [comment]
@leeebo Thank you for the fast review. I fixed all issues. The license should be right, we are using it in all files in BSP.

I reworked buttons for new API too and I am working on changes in CI for testing only some examples in IDF4.4

File: esp-box/esp-box_idf4.c
Comment (espzav) [reply esp-box/esp-box_idf4.c:4]: We are using same license in all BSP files.

Comment (igrr) [reply esp-box/esp-box_idf4.c:4]: @espzav In that case, we should change it to Apache 2.0 license, since you have an apache license in esp-box component: https://github.com/espressif/esp-bsp/blob/c6ae6ebf02bbbadaf583503cea72a7912c2b3bb5/esp-box/LICENSE

Comment (espzav) [reply esp-box/esp-box_idf4.c:4]: @igrr Thanks for clarify. I see, that we have it same in all BSPs. We should consider it. I will create ticket and I will make it. But problem was, I don't understand licesnes too much. I didn't know, which header should be there. 
Thank you for help.

**Comment by donghengqaz** [comment]
API `esp_lcd_panel_disp_on_off` is not included in esp-idf v4.4.3, I see in other code, using `#if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(5, 0, 0)` to chose which API is compiled.

**Comment by donghengqaz** [comment]
I test in esp-idf v4.4.3, but find it may block in lvgl code `lvgl/src/core/lv_refr.c` boss in single and double buffer mode.

```
if(disp_refr->driver->wait_cb) disp_refr->driver->wait_cb(disp_refr->driver);
```

**Comment by espzav** [comment]
@donghengqaz Thank you for your inputs. Do you really need support of IDF 4.4.3 ? We want to test only latest release of IDF4.4

**Comment by donghengqaz** [comment]
> @donghengqaz Thank you for your inputs. Do you really need support of IDF 4.4.3 ? We want to test only latest release of IDF4.4

No, I planed to update application to support IDF v4.4.5, but now it is v4.4.3. And you had better set the IDF version requirement to be 4.4.5.

**Comment by espzav** [comment]
> > @donghengqaz Thank you for your inputs. Do you really need support of IDF 4.4.3 ? We want to test only latest release of IDF4.4
> 
> No, I planed to update application to support IDF v4.4.5, but now it is v4.4.3. And you had better set the IDF version requirement to be 4.4.5.

Ok, we will make it for IDF 4.4.5

**Comment by donghengqaz** [comment]
> Besides this, we need option in Kconfig to configure LVGL task's stack size.

@espzav Please add Kconfig to configure LVGL task's stack size, and I will switch your new version if it is done.

**Comment by espzav** [comment]
> > Besides this, we need option in Kconfig to configure LVGL task's stack size.
> 
> @espzav Please add Kconfig to configure LVGL task's stack size, and I will switch your new version if it is done.

I added new function `lv_disp_t *bsp_display_start_with_config(const bsp_display_cfg_t *cfg)` int BSP, where you can put `lvgl_port_cfg_t` with `task_priority`, `task_stack`, `task_affinity`, `task_max_sleep_ms` and `timer_period_ms`. You can use this function instead of `bsp_display_start()`.

File: examples/display_audio_photo/main/idf_component.yml
Comment (igrr) [inline examples/display_audio_photo/main/idf_component.yml:3]: I think you can completely remove `idf: ` line from `main/idf_component.yml`, since the example itself doesn't depend on any particular IDF version; It's enough to specify the dependency in the BSP component

File: esp-box/esp-box_idf4.c
Comment (espzav) [reply esp-box/esp-box_idf4.c:4]: Fixed in #195

**Comment by donghengqaz** [comment]
> > > Besides this, we need option in Kconfig to configure LVGL task's stack size.
> > 
> > 
> > @espzav Please add Kconfig to configure LVGL task's stack size, and I will switch your new version if it is done.
> 
> @donghengqaz It's your PR, mater group has the same requirement ? #190

I don't know more about this, I create an [issue](https://github.com/espressif/esp-bsp/issues/151) long long ago, besides this I find my blocking issue is in SPI bus lock function, not related to BSP.

**Comment by espzav** [comment]
Please all, don't forget to approve, when the PR is OK. I cannot merge it till I get at least one approve ;-)

**Review by leeebo — APPROVED** [review]
LGTM!

File: .github/PULL_REQUEST_TEMPLATE/pr_template_bsp.md
Comment (igrr) [inline .github/PULL_REQUEST_TEMPLATE/pr_template_bsp.md:10]:
```suggestion
- [ ] BSP was [added to SquareLine](https://github.com/espressif/esp-bsp/tree/master/SquareLine/common)
```

File: .github/PULL_REQUEST_TEMPLATE/pr_template_bsp.md
Comment (igrr) [inline .github/PULL_REQUEST_TEMPLATE/pr_template_bsp.md:11]:
```suggestion
- [ ] New BSP supports IDF v4.4.x and later releases
```

File: esp-box/idf_component.yml
Comment (igrr) [inline esp-box/idf_component.yml:10]: According to https://github.com/espressif/esp-bsp/pull/192#issuecomment-1643322561, you should change this to 4.4.5:
```suggestion
  idf: ">=4.4.5"
```

File: esp-box/include/bsp/esp-box.h
Comment (igrr) [inline esp-box/include/bsp/esp-box.h:175]: Is it a breaking change to remove this function?

**Review by igrr — COMMENTED** [review]
Haven't reviewed the source code changes in detail, just a few small comments about the infrastructure.

File: esp-box/idf_component.yml
Comment (espzav) [reply esp-box/idf_component.yml:10]: Sure. I think, there should be one CI test for exactly IDF 4.4.5 and second one for latest release IDF 4.4 
Are you agree?

File: esp-box/include/bsp/esp-box.h
Comment (espzav) [reply esp-box/include/bsp/esp-box.h:175]: Yes. The whole audio changes to `esp_codec_dev` are breaking changes.

**Comment by espzav** [comment]
@igrr Thank you for review. I left some answers to your comments.

File: esp-box/idf_component.yml
Comment (igrr) [reply esp-box/idf_component.yml]: This is not related to CI test, i think... we can keep CI running for release/v4.4, just need to make sure that this component won't be installed if IDF version is <= 4.4.4 — preventing the issue which @donghengqaz ran into.

File: esp-box/include/bsp/esp-box.h
Comment (igrr) [reply esp-box/include/bsp/esp-box.h:175]: In that case, why is the major version of the component not incremented? https://semver.org/#summary

**Review by tore-espressif — APPROVED** [review]
LGTM!

**Pull request merged**
